corehq/apps/app_manager/views/apps.py

```diff
@@ -612,15 +612,17 @@ def import_app(request, domain):
             messages.error(request, _("You must upload the app source file."))
             valid_request = False
 
-        if not valid_request:
-            return render(request, template, {'domain': domain})
-
         try:
-            source = json.loads(file.read())
+            source = json.load(file)
```

Comment:
This will fail with a different error message if `file` does not have a `read` attribute (which seems like it could happen based on the validation above.

I think you should still exit early if validation has already failed and also exit if the JSON read fails.

Reply:
Thanks, updated now.